qa/L0_trace/opentelemetry_unittest.py

```diff
@@ -40,29 +41,54 @@
 
 class OpenTelemetryTest(tu.TestResultCollector):
     def setUp(self):
+        # Extracted spans are in json-like format, thus data needs to be
+        # post-processed, so that `json` could accept it for further
+        # processing
         with open("trace_collector.log", "rt") as f:
```

Comment:
What code generates/outputs this "trace_collector.log" ? Is it Triton code, or open telemetry built-in code?

Reply:
This line: https://github.com/triton-inference-server/server/pull/6063/files#diff-27e9ca4cd8f292f0e5a94bbabb243e40fe0d47cd55164a0cb49ae6e093c9fbe7R778

It is extracted from server logs into a separate file

Reply:
The code is generated by OStream exporter (OpenTelemetry side) : https://github.com/open-telemetry/opentelemetry-cpp/blob/main/exporters/ostream/src/span_exporter.cc#L73

Reply:
In their python instrumentation, they do export spans in `json`: [Example](https://opentelemetry.io/docs/instrumentation/python/automatic/example/). To do this they use [`ConsoleExporter`](https://github.com/open-telemetry/opentelemetry-python/blob/main/docs/examples/auto-instrumentation/server_programmatic.py#L27), defined [here](https://github.com/open-telemetry/opentelemetry-python/blob/main/opentelemetry-sdk/src/opentelemetry/sdk/trace/export/__init__.py#L499). As you can see, they invoke `to_json` on every processed span, defined [here](https://github.com/open-telemetry/opentelemetry-python/blob/main/opentelemetry-sdk/src/opentelemetry/sdk/trace/__init__.py#L455). Thus, they don't have an explicit parser, that can help us here.

Reply:
@rmccorm4  and @GuanLuo how about I create a request on opentelemetry-cpp github to add an option for OStream to print spans in json format and will monitor. As soon as they add this, I'll change this part of the test.

Reply:
Sharing the json file contrib mentioned offline for future reference: https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/exporter/fileexporter, but as concluded it's not shipped with the standard set of exporters out of the box, probably not worth the effort figuring out how to integrate it.

> how about I create a request on opentelemetry-cpp github to add an option for OStream to print spans in json format and will monitor. As soon as they add this, I'll change this part of the test.

No harm in raising the request to add flag for ostream exporter, go for it. After poking around, it doesn't seem like there are quick of easy options for improving the parsing/ingestion of these logs.